packages/core/src/amazonq/webview/messages/messageDispatcher.ts

```diff
@@ -86,12 +88,19 @@ export function dispatchWebViewMessagesToApps(
         }
 
         if (msg.type === 'error') {
-            const event = msg.event === 'webview_load' ? telemetry.webview_load : telemetry.webview_error
-            event.emit({
-                webviewName: 'amazonqChat',
-                result: 'Failed',
-                reasonDesc: msg.errorMessage,
-            })
+            if (msg.event === 'toolkit_didLoadModule') {
+                telemetry.toolkit_didLoadModule.emit({
```

Comment:
Why does the `webview_error` metric exist? Shouldn't it in theory mirror the failures of `toolkit_didLoadModule` or are there non-error reasons a webview fails to load?

If we have a separate metric for error, shouldn't we emit it in this case since we still received an error from the webview?

Reply:
I'm thinking of dropping `webview_error` and creating something like `toolkit_moduleError`. This will capture any errors that happen **after** loading has happened, anything before would be captured in `toolkit_didLoadModule`.

I have it as a TODO to deprecate `webview_error`, this will also allow us to deal with different field names (`module` vs `webviewName`)

Reply:
> `toolkit_moduleError`. This will capture any errors that happen **after** loading has happened, anything before would be captured in `toolkit_didLoadModule`.

? errors should be part of all metrics. There should not be a separate "foo_error" metric.